libs/feature/editor/src/lib/components/record-form/form-field/form-field-overviews/form-field-overviews.component.ts

```diff
@@ -53,7 +54,26 @@ export class FormFieldOverviewsComponent {
   handleFileChange(file: File) {
     this.uploadProgress = 0
     this.uploadSubscription = this.platformService
-      .attachFileToRecord(this.metadataUuid, file)
+      .getRecordAttachments(this.metadataUuid)
+      .pipe(
+        map((recordAttachement) => recordAttachement.map((r) => r.fileName)),
+        switchMap((fileNames) => {
+          let fileToUpload = file
+
+          if (fileNames.includes(file.name)) {
+            const fileNameParts = file.name.split('.')
+            const extension = fileNameParts.pop()
+            const baseName = fileNameParts.join('.')
+            const newFileName = `${baseName}_${Date.now()}.${extension}`
+
+            fileToUpload = new File([file], newFileName, { type: file.type })
+          }
+          return this.platformService.attachFileToRecord(
+            this.metadataUuid,
+            fileToUpload
+          )
+        })
+      )
```

Comment:
This should be abstracted away in `platformService.attachFileToRecord` IMO, to avoid repeating it and having too much responsibility in the components.

That means this method should probably return the full path of the file on upload success.

Reply:
Also I would put the filename manipulation logic in a separate util, so has to be able to test it a bit more. For example this code will probably fail on a filename without dots, right?

Reply:
Yes, but what do you want me to do for it ? Make sure that the code still work if the uploaded file has no extension ?